src/charm.py

```diff
@@ -82,6 +83,21 @@ def _on_install(self, event: ops.InstallEvent) -> None:
         self.sshdebug.update_relation_data(host=str(self.state.ip_addr), fingerprints=fingerprints)
         self.unit.status = ops.ActiveStatus()
 
+    def _on_update_status(self, _: ops.UpdateStatusEvent) -> None:
+        """Check the health of the workload and restart if necessary."""
+        if not tmate.is_running():
+            logger.error("tmate-ssh-server is not running. Will restart.")
+
+            tmate.start_daemon(address=str(self.state.ip_addr))
+
+            logger.info("Removing stopped containers.")
+            try:
+                tmate.remove_stopped_containers()
+            except tmate.DockerError:
+                logger.exception("Failed to remove stopped containers.")
```

Comment:
Should the charm go into blocked state/ error after this? It looks like it will go into active state at the moment which doesn't seem right given that the tmate service would not be running?

Reply:
This is just to remove the stopped Docker containers, a sort of cleanup to prevent disk exhaustion. I don't think this should block the charm.